pom.xml

```diff
@@ -832,6 +846,224 @@ Arrays, Maps or Strings. The iteration protocol was inspired by the Smalltalk co
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <id>spotless-check</id>
+
+            <properties>
+                <spotless.check.skip>false</spotless.check.skip>
+            </properties>
+
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.diffplug.spotless</groupId>
+                        <artifactId>spotless-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>check</goal>
+                                </goals>
+                                <phase>validate</phase>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <profile>
+            <id>spotless-apply</id>
+
+            <properties>
+                <spotless.check.skip>false</spotless.check.skip>
+            </properties>
+
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.diffplug.spotless</groupId>
+                        <artifactId>spotless-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>apply</goal>
+                                </goals>
+                                <phase>verify</phase>
```

Comment:
The apply goal is meant to be run locally, to auto-format code. By itself, it doesn't do anything. We need to activate at least two profiles together to perform some auto-formatting. For example, `mvn spotless:apply --activate-profiles 'spotless-apply,spotless-java-sort-imports'` would fix up imports locally. A similar command could run from CI, just swapping in the word check for apply.